test/datalog_tests/simple.dl

```diff
@@ -681,3 +681,24 @@ Allocation(name, id, x) :-
     Allocated(name, xs),
     var allocation = FlatMap(allocate_u32(xs, ids, (32'd1 << 32'd24) - 32'd1)),
     (var id, var x) = allocation.
+
+/* Ref test */
+
+input relation Referenced(x: bool, y: Option<string>)
+
+output relation Referee(r: Ref<Referenced>)
+
+Referee(ref_new(r)) :-
+    r in Referenced(_, Some{_}).
+
+output relation Filtered(r: Ref<Referenced>)
+
+Filtered(r.r) :- r in Referee(.r = &Referenced{.x = true}).
```

Comment:
Is equality for references well-defined? I suspect they behave like pointers. But this means you cannot just copy them.

Reply:
I.e., if you delete a value and insert it back again you will have the same value in the relation, but the other relation will have a different reference. I will have to think about the semantics of this.

Reply:
The current semantics compares values inside references. In the scenario you described, there should still only be one reference, but there theoretically exist more complex dataflows where this is indeed possible. In any case, multiple references should not affect the functionality; only memory footprint.